src/main/connection.cpp

```diff
@@ -27,36 +25,25 @@ namespace main {
 Connection::Connection(Database* database) {
     assert(database != nullptr);
     this->database = database;
-    clientContext = std::make_unique<ClientContext>();
-    transactionMode = ConnectionTransactionMode::AUTO_COMMIT;
+    clientContext = std::make_unique<ClientContext>(database);
 }
 
-Connection::~Connection() {
-    if (activeTransaction) {
-        database->transactionManager->rollback(activeTransaction.get());
-    }
-}
+Connection::~Connection() {}
 
 void Connection::beginReadOnlyTransaction() {
-    std::unique_lock<std::mutex> lck{mtx};
```

Comment:
Can you add a comment that those apis are deprecated. Developers should avoid using these

Reply:
Added an issue